apps/web/src/components/Basenames/RenewalForm/index.tsx

```diff
@@ -39,19 +37,23 @@ export default function RenewalForm({ name }: { name: string }) {
     [connectedChain],
   );
 
-  const nameData = useMemo(() => {
-    return namesData?.data.find((n) => n.domain === formatBaseEthDomain(name, basenameChain.id));
-  }, [basenameChain.id, name, namesData?.data]);
-
-  const formattedExpirationDate = useMemo(() => {
-    if (!nameData?.expires_at) return undefined;
-    const date = new Date(nameData.expires_at);
-    return date.toLocaleDateString('en-US', {
-      month: '2-digit',
-      day: '2-digit',
-      year: 'numeric',
-    });
-  }, [nameData?.expires_at]);
+  const fetchExpirationDate = useCallback(async () => {
+    try {
+      const formattedName = formatBaseEthDomain(name, basenameChain.id);
+      const expiresAt = await getBasenameNameExpires(formattedName);
```

Comment:
By using this, we're able to fetch the expiration date for _any_ Basename, not just the one you own